api/notifications/signals.py

```diff
@@ -24,16 +24,21 @@ def post_save_user(sender, instance, created, **kwargs):
 @receiver(post_save, sender=Action, dispatch_uid="ActionSave")
 def post_save_action(sender, instance, using=None, **kwargs):
     """Create appropriate notifications when an action changes."""
+
+    # Track what has changed before enqueuing
+    # As the instance inside the worker won't have changed
+    changed = instance.tracker.changed()
+
     # Create the notification only after the save of the action
     # is commited, to ensure action will be found to satisfy
     # the foreign key on the notifications table
     transaction.on_commit(
         lambda: django_rq.enqueue(
-            notifications.create_action_notifications, instance, result_ttl=0
+            notifications.create_action_notifications, instance, changed, result_ttl=0
         ), using=using)
 
 
-@receiver(post_save, sender=Notification, dispatch_uid="NotificationSave")
+@ receiver(post_save, sender=Notification, dispatch_uid="NotificationSave")
```

Comment:
I assume this space is accidental?

Reply:
Removed

Reply:
ooh, yeah, hadn't noticed it, good catch :)